src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -300,18 +300,18 @@ public virtual VerifyResult VerifyStateDependent(StoreView snapshot, Transaction
             return VerifyResult.Succeed;
         }
 
-        public virtual VerifyResult VerifyStateIndependent()
+        public virtual VerifyResult VerifyStateIndependent(StoreView snapshot)
```

Comment:
Why add `snapshot`? `VerifyStateIndependent` should keep it state independent. I see we have imported `base_exec_fee`,    we should move the fee check of state independent to state dependent. 

I think we can merge this PR, which can only focus on the opcode fee.  And we can send another two PRs:
1) one for syscall fee
2) another for VerifyStateIndependent.

Reply:
> we should move the fee check of state independent to state dependent.

The execution fee for state-independent scripts can only be calculated during execution (in general) and to calculate it we need `base_exec_fee` to be applied.

Reply:
Since 'base_exec_fee' is not changed very often, we can reverify the whole transactions when `base_exec_fee`  is modified.

Reply:
see https://github.com/neo-project/neo/pull/2054

Reply:
Yeah, that would allow to remove dependency on `snapshot` here.